jupyter_server/services/events/handlers.py

```diff
@@ -75,6 +79,45 @@ def on_close(self):
         self.event_bus.handlers.remove(self.logging_handler)
 
 
+class EventHandler(APIHandler):
+    """REST api handler for events"""
+
+    auth_resource = AUTH_RESOURCE
+
+    @property
+    def event_bus(self):
+        """Jupyter Server's event bus that emits structured event data."""
+        return self.settings["event_bus"]
+
+    @web.authenticated
+    @authorized
+    async def post(self):
+        payload = self.get_json_body()
+        if payload is None:
+            raise web.HTTPError(400, "No JSON data provided")
+
+        try:
+            if "timestamp" in payload:
+                timestamp = datetime.strptime(payload["timestamp"], "%Y-%m-%d %H:%M:%S")
```

Comment:
I think it would be a good idea to include microseconds in this kind of timestamp since we should be producing sub-second events.

I would also recommend recording the UTC offset so that "rollup/analysis" applications can operate on data from different timezones.  Using an offset is probably more user-friendly wrt today's Jupyter users than going with UTC-based timestamps directly (although more difficult for analysis).

```suggestion
                timestamp = datetime.strptime(payload["timestamp"], "%Y-%m-%d %H:%M:%S.%f %z")
```

Reply:
Thanks for that suggestion. I will update in the next push.

Reply:
@kevin-bates I ended up removing the microseconds from the date format because EventLog is dropping this in the record_event method; `isoformat` by default drops the microseconds. 
https://github.com/jupyter/telemetry/blob/master/jupyter_telemetry/eventlog.py#L237

Let me know if we can handle this in a better way.